parla/task_runtime.py

```diff
@@ -1126,48 +1123,45 @@ def check_resources_availability(self, d: Device, resources: ResourceDict):
         :param resources: The resources to deallocate.
         """
         logger.debug("[ResourcePool] Acquiring monitor in check_resources_availability()")
-        with self._monitor:
```

Comment:
Are we *sure* removing this monitor is safe?

Reply:
I suppose it is if *only* the scheduling thread makes this call.

Reply:
I checked it by printing a thread and it was called only by the scheduler. But let me double check this.

Reply:
It can also be called by any thread that creates a PArray: https://github.com/ut-parla/Parla.py/blob/d215415572eb0b827529e5d142f6dd30b0d6c030/parla/parray/core.py#L52